Clarify console access via localhost (#168)

ShaunCurrier opened the pull request:
Also, minor editing for clarity and ease of use.
Signed-off-by: Shaun Currier <shaun.currier@gmail.com> (github: ShaunCurrier)

Fixes #168

ThomDietrich commented:
Hey Shaun,
thanks for the contribution! Is the PR finished or are you also going to address the other open issues in #168, #298 and #299 ?

ShaunCurrier commented:
Hi Thom,
Whoops, I didn't see those related threads that came after the issue I opened.

I'm now looking into addressing everything, so consider this a work in progress.

ThomDietrich commented:
Okay ;) If you are not sure when you'll find the time, we can also merge right away and address other issues in new PRs. Just so you know

ThomDietrich commented on administration/console.md:
```diff
@@ -15,15 +15,26 @@ The console offers the option to:
-The manually start using shell/batch script ends directly in the console.
+Accessing the console depends on how openHAB was started.
+When started manually using shell or batch script, openHAB naturally transitions directly to the console.
+When started as a service, the console can be reached via local SSH.  
```
Even explained this good, users might not know which path they are on. e.g openHABian user might not know the meaning of "service". Maybe it would be wise to mention the "installed from package repository" here somehow.

I'd remove "local" here and instead link SSH, users might not know what it is.

ShaunCurrier replied:
I agree.  Shall I break the "Accessing the Console" into two sections and give a short list helping users know what method they probably started openHAB by? I'd keep it as concise as possible. Then we could add and remove from the lists as things change and more install methods or purpose built OS's are added.

Can you clarify what you mean by, "instead link SSH?"

ThomDietrich replied:
I'd not start to give a list with all possible OS's or something like that. The user is still encouraged to think for him/herself 😄  

Just one idea:

> When started in interactive mode using the provided command line scripts, openHAB naturally transitions directly to the console prompt.
> When started as a service (e.g. when installed from our package repository), openHAB is running as background process.
> In both cases the console can be reached via Secure Shell connection ([SSH](https://en.wikipedia.org/wiki/Secure_Shell)).

One other thing: I think most users nowadays are using some kind of openHAB service and will connect to the Console via SSH. Maybe we should move the first sentence further down anyhow?

ShaunCurrier replied:
Agreed, I will do this.

ShaunCurrier replied:
Done.

ThomDietrich commented on administration/console.md:
```diff
@@ -15,15 +15,26 @@ The console offers the option to:
+If this happens, simply try connecting again until successful.
+
+By default, the console can only be reached from the machine running openHAB.
+To change this, see "Bind Console to All Interfaces," below.
```
Please insert a link here

ShaunCurrier replied:
I'm unfamiliar with the markdown parser that Jekyll uses, can you confirm that this is the proper way to do an anchor link?

`[Bind Console to all Interfaces](#bind-console-to-all -interfaces`

If I should just bit the bullet and install the doc build tools so I can test myself, feel free to tell me.

ThomDietrich replied:
`[Bind Console to all Interfaces](#bind-console-to-all-interfaces)` - yes, that's correct.
This anchor is already posted at a few places. Otherwise I'd actually go one step further and define a shorter headline title unrelated id, see [here at the bottom](https://raw.githubusercontent.com/openhab/openhab-docs/gh-pages/installation/openhabian.md) for an example.

If you intend to continue helping out (which would be **A-MA-ZING**) you should get a build environment, yes. However it's not a bullet if you ask me, you can use the virtual machine I've provided. https://github.com/openhab/openhab-docs/blob/gh-pages/README.md#alternative-vagrant-vm

ShaunCurrier replied:
Thanks, I'll replace this.

ThomDietrich commented:
@ShaunCurrier I almost forgot to bring https://github.com/openhab/openhab-docs/issues/345 to the game

ShaunCurrier commented:
I think I've got it to a pretty good place now.  Only thing that I saw discussed that was not done was how to use certs instead of a password.  I'm going to leave that to someone else.

I addressed:
* Linking to the "bind to all interfaces" section from another section
* How to change the console port
* How to change password
* Broke out the various console settings modifications into separate headers for easier navigation (is it too much?)
* Added tables of different systems and where the console settings files might be found (thanks @chriscolden for most of this work)
* Made the console connection less surprising to the inexperienced via a walkthrough
* Refreshed the successful console connection example
* Incorporated Thom's initial feedback

One last thing I have left to do is install the vagrant and proof read the whole thing.  I'll plan to do that.

@ThomDietrich, please review and provide feedback.

Besides any feedback you might give, specifically, how do you want me to clean up all these commits before merging?

ThomDietrich commented:
Very nice! I'll take my time to have a look at this PR and the one by @stephenbonar this evening.
After that I'll finally comment on your thread about myopenhab (which I did not forget!)

Great work guys ;)

ThomDietrich commented:
@ShaunCurrier great additions! I've added a commit myself, including a few details, corrections, formatting....
I've reduced the Password chapter back down to one part, I really don't see why hashed or not should make a difference when wanting to change the password.
The second thing I decided to remove was the table by @chriscolden. We shouldn't start to try and count all possible setup variations each time we talk about files. Instead I wanted to [introduce unique placeholders](https://github.com/openhab/openhab-docs/issues/189) into the documentation. I've not come far with that yet but this article is now the first one to use one.

It would be great if you could review my changes just as I did yours. If you are in any way not pleased with my changes, let me know! I'm neither without fault nor the boss here 😉

ShaunCurrier commented:
@ThomDietrich, looks really good and I agree with all your edits.

ShaunCurrier commented:
I did put a few comments in that you should look at, but after that I vote go with it.

chriscolden commented:
Love the idea of placeholders. It would certainly resolve the need to define where all the files are each time. Instead just reference them. Alternatively add a files location section into the docs. Either way it would be very useful.

ThomDietrich commented:
Finally merged. Sorry for the delay.
@ShaunCurrier @chriscolden thanks again! Would be happy to see new contributions by you soon ;)
